src/Microsoft.Extensions.Localization/ResourceManagerStringLocalizerFactory.cs

```diff
@@ -50,6 +50,29 @@ public class ResourceManagerStringLocalizerFactory : IStringLocalizerFactory
             }
         }
 
+        protected virtual string GetKey(TypeInfo typeInfo)
+        {
+            return GetKey(typeInfo, _hostingEnvironment.ApplicationName, _resourcesRelativePath);
+        }
+
+        protected virtual string GetKey(TypeInfo typeInfo, string location, string resourcesRelativePath)
+        {
+            return string.IsNullOrEmpty(resourcesRelativePath)
```

Comment:
Could you add a comment on how these keys become in the end after the transformation(also in the other overload)? not immediately clear what this is doing unless i spend some time calculating the parts.